src/formatDate.js

```diff
@@ -8,7 +8,54 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYearIndex = getYearIndex(fromFormat);
+  const newYearIndex = getYearIndex(toFormat);
+  const oldYearLength = fromFormat[oldYearIndex].length;
+  const newYearLength = toFormat[newYearIndex].length;
+  const oldJoin = fromFormat[3];
+  const newJoin = toFormat[3];
+  const newDate = date.split(oldJoin);
+  const reorderedDate = [];
+  const oldYear = newDate[oldYearIndex];
+
+  newDate[oldYearIndex] = convertYearFormat(
+    oldYear,
+    oldYearLength,
+    newYearLength,
+  );
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (i === newYearIndex) {
+      reorderedDate[i] = newDate[oldYearIndex];
+    } else {
+      const index = fromFormat.indexOf(toFormat[i]);
+
+      reorderedDate[i] = newDate[index];
+    }
+  }
+
+  return reorderedDate.join(newJoin);
+}
+
+function getYearIndex(index) {
+  const shortYear = index.indexOf('YY');
+  const longYear = index.indexOf('YYYY');
+
+  return shortYear !== -1 ? shortYear : longYear;
+}
+
+function convertYearFormat(year, from, to) {
+  const twoLastYearDigits = '30';
```

Comment:
For this type of constant values, we use uppercase letters.

```suggestion
  const TWO_LAST_YEAR_DIGITS = '30';
```